src/epics/router.ts

```diff
@@ -185,12 +185,14 @@ const drillDown: Epic = (action$, store) => action$.ofType(types.drillDown)
                                 const parsedFilters = parseFilters(filters[bcName])
                                 parsedFilters.forEach((item) => {
                                     store.dispatch($do.bcAddFilter({bcName, filter: item}))
+                                    store.dispatch($do.bcForceUpdate({bcName}))
                                 })
                                 if (!diff.length) {
                                     store.dispatch($do.bcForceUpdate({bcName}))
                                 }
                             } else {
                                 store.dispatch($do.bcRemoveAllFilters({bcName}))
+                                store.dispatch($do.bcForceUpdate({bcName}))
```

Comment:
* You are calling `bcForceUpdate` in each branch, => it should be one call outside of branches
* You are pushing `bcForceUpdate` for the same BC on each filter, meaning if you filter one business component by five different columns, this bc will fetch data five times and rerenders more than five times.
* overall it doesn't seem like a proper place to fix it. From the #374 I grasp that if we have view1 and view2 with the same widget, than filtration drilldown from view1 to view2 will apply new filters, but will not refetch the data with new filters. It's unlikely as `changeLocation` by itself causes updates (a little more likely if we talking about filtration drilldown onto the same view), but either way I suspect it should be handled the same way as we handle "same view, different cursors" scenario:
https://github.com/tesler-platform/tesler-ui/blob/1c94589c7d9c1ff06a5220f775302e44f97a9956/src/epics/router.ts#L86-L96

Reply:
I believe that it necessary to update the BC in this epic.
It is true that if there are several filters, then the same BC will be updated the same number of times.
I fixed this by adding a change that would update only once each BC by which filter or sort with drilddown.


When `changeLocation` action is executed, we will no longer have an idea of ​​which BC filter was in the drilledown and which BC needs to be updated.
If try to associate a BC update with filter or sort in store, then problems will arise when drillDown with remove filter, or with filters for other BC that do not participate in drillDown with filtering.